src/nvim/tui/tui.c

```diff
@@ -1288,6 +1292,20 @@ static void tui_option_set(UI *ui, String name, Object value)
   }
 }
 
+static void tui_term_osc_cmd(UI *ui, String cmd)
+{
+  char *seq = xmalloc(cmd.size + 4);
+  snprintf(seq, cmd.size + 4, "\033]%s\007", cmd.data);
```

Comment:
There should be no need to hardcode specifically this sequence form in the ui event itself, rather we should have a generic `term_output` event which just forwards any string unmodified.